pom.xml

```diff
@@ -212,6 +212,10 @@
             </path>
           </annotationProcessorPaths>
           <release>8</release>
+          <compilerArgs>
```

Comment:
Consider adding:
```xml
<failOnWarning>true</failOnWarning>
```
This will require some source adjustments to pass the build, though.

Reply:
> Consider adding:
> 
> ```
> <failOnWarning>true</failOnWarning>
> ```
> 
> This will require some source adjustments to pass the build, though.

Do we run `failOnWarning` for compilation on any java project or is this a new initiative for googleapis (I'm not against it, just wondering)?

If we add this, perhaps I can create a new GH issue and make the relevant source changes in a separate PR.

Reply:
It's a good idea but there are a lot of preexisting warnings currently, some of them require us to change the surface and bump major versions, so I wouldn't add it as part of this PR.

> If we add this, perhaps I can create a new GH issue and make the relevant source changes in a separate PR.

+1. Create a separate GH issue and add a `semver: major` label sounds good to me.

Reply:
Also people have tried to fix the warnings but we didn't want to because of the major version bump concern, see this [PR](https://github.com/googleapis/gax-java/pull/1633).

Reply:
Created an GH issue and linked the PRs/ Issues there: https://github.com/googleapis/gax-java/issues/1871. I think we can track progress there.

Reply:
Thanks, @lqiu96